packages/discord-request/test/util/mock-fetch.ts

```diff
@@ -0,0 +1,35 @@
+import { MockAgent, setGlobalDispatcher } from "undici";
+
+export function getMockClient() {
+	const mockAgent = new MockAgent();
```

Comment:
do you really want a new agent every time?
could / should store this at the file level so its only created once per process